pkg/clusters/types/gke/builder.go

```diff
@@ -176,3 +194,67 @@ func (b *Builder) Build(ctx context.Context) (clusters.Cluster, error) {
 
 	return cluster, nil
 }
+
+// createCluster creates the GKE cluster asynchronously.
+func (b *Builder) createCluster(ctx context.Context, req *containerpb.CreateClusterRequest, mgrc *container.ClusterManagerClient, createdByID, authToken string) error {
+	// createSubnet is currently only available via gcloud CLI:
+	// https://github.com/googleapis/google-cloud-go/issues/7219
+	if b.createSubnet {
+		if err := b.createClusterUsingCLI(ctx, req, createdByID, authToken); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	_, err := mgrc.CreateCluster(ctx, req)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (b *Builder) createClusterUsingCLI(ctx context.Context, req *containerpb.CreateClusterRequest, createdByID, authToken string) error {
+	tokenFile, err := os.CreateTemp("", "gcloud-token-")
+	if err != nil {
+		return fmt.Errorf("failed to create a temporary file for gcloud token: %w", err)
+	}
+	defer func() {
+		_ = os.Remove(tokenFile.Name())
+	}()
+	if _, err := io.WriteString(tokenFile, authToken); err != nil {
+		return fmt.Errorf("failed to write a token to the temporary file: %w", err)
+	}
+
+	//nolint:gosec
+	cmd := exec.CommandContext(ctx, "gcloud", "container", "clusters", "create", req.Cluster.Name,
+		`--access-token-file`, tokenFile.Name(),
+		`--project`, b.project,
+		`--region`, b.location,
+		`--create-subnetwork`, ``,
+		`--enable-ip-alias`,
+		`--num-nodes`, `1`,
+		`--cluster-version`, req.Cluster.InitialClusterVersion,
+		`--addons`, `HorizontalPodAutoscaling`,
```

Comment:
Do we really need the HPA addon?

Reply:
Yeah, we don't need it. I was just struggling with passing an empty set of addons and that was a nasty workaround for not using GKE ingress controller. It's fixed now: 4298d6c